File: stripe/src/main/java/com/stripe/android/view/AddPaymentMethodRowView.java
package com.stripe.android.view;

import android.app.Activity;
import android.support.annotation.IdRes;
import android.support.annotation.LayoutRes;
import android.support.annotation.NonNull;
import android.view.View;
import android.widget.FrameLayout;

abstract class AddPaymentMethodRowView extends FrameLayout {
    AddPaymentMethodRowView(@NonNull final Activity activity,
                            @LayoutRes int layoutId,
                            @IdRes int idRes,
                            @NonNull final AddPaymentMethodActivityStarter.Args args) {
        super(activity);
        inflate(activity, layoutId, this);
        setId(idRes);

        setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(@NonNull View view) {
                new AddPaymentMethodActivityStarter(activity).startForResult(args);
            }
        });
    }
}

File: stripe/src/main/java/com/stripe/android/view/AddPaymentMethodRowView.kt
package com.stripe.android.view

import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import android.widget.FrameLayout
import com.stripe.android.databinding.AddPaymentMethodRowBinding

internal class AddPaymentMethodRowView private constructor(
    context: Context
) : FrameLayout(context) {

    private val viewBinding = AddPaymentMethodRowBinding.inflate(
        LayoutInflater.from(context),
        this,
        true
    )

    internal val label = viewBinding.label

    init {
        layoutParams = ViewGroup.LayoutParams(LayoutParams.MATCH_PARENT, LayoutParams.WRAP_CONTENT)
        isFocusable = true
        isClickable = true
    }
}
